test/system/170-run-userns.bats

```diff
@@ -179,3 +179,27 @@ EOF
     # gid not mapped
     run_podman run --rm --uidmap 0:0:1000 --gidmap 0:1:1000 $IMAGE true
 }
+
+# bats test_tags=ci:parallel
+@test "podman --userns=ns:<path> join existing user namespace" {
+    skip_if_not_rootless
+    # Test for issue #27148: --userns=ns:<path> should not add dummy mappings
+
+    local cname="userns_source_$(random_string 8)"
+
+    # Create a long-running container with custom user namespace mappings
+    run_podman run -d --name $cname \
+        --userns=keep-id \
+        $IMAGE top
+
+    run_podman inspect --format '{{.State.Pid}}' $cname
+    local pid=$output
+    local userns_path="/proc/$pid/ns/user"
+
+    test -e "$userns_path"
+
+    run_podman run --rm --userns=ns:$userns_path $IMAGE echo "success"
+    assert "$output" == "success" "Should be able to join existing user namespace"
```

Comment:
The echo doesn't actually test that we successfully joined the same userns though. If userns is a NOP

see the test in test/system/195-run-namespaces.bats as example, you should run something like `readlink /proc/self/ns/user` and then compare that against the `podman exec $cname readlink /proc/self/ns/user` output to ensure they do in fact use the same userns, we could also compare `cat /proc/self/uid_map` for good measures